storage/onpremise/mongodb/variables.tf

```
@@ -45,44 +45,39 @@ variable "persistent_volume" {
   type = object({
     access_mode         = optional(list(string), ["ReadWriteMany"])
     reclaim_policy      = optional(string, "Delete")
-    storage_provisioner = string
-    volume_binding_mode = string
+    storage_provisioner = optional(string, "")
+    volume_binding_mode = optional(string, "")
     parameters          = optional(map(string), {})
 
     # Resources for PVC
-    resources = object({
-      limits = object({
+    resources = optional(object({
+      limits = optional(object({
         storage = string
-      })
-      requests = object({
+      }))
+      requests = optional(object({
         storage = string
-      })
+      }))
+    }),
+    {
+      requests = {
+        storage = "8Gi"
+      }
```

Are you able to deploy without a request? If so, you can just let it null or empty.

```suggestion
    {}
```

---

If requests.storage is explicitly set to null or empty string the deployment fails.

---

You're right